clusterloader2/pkg/measurement/common/system_pod_metris.go

```diff
@@ -0,0 +1,127 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package common
+
+import (
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/klog"
+	"k8s.io/perf-tests/clusterloader2/pkg/measurement"
+	"k8s.io/perf-tests/clusterloader2/pkg/util"
+)
+
+const (
+	systemPodMetricsName            = "SystemPodMetrics"
+	systemNamespace                 = "kube-system"
+	systemPodMetricsEnabledFlagName = "systemPodMetricsEnabled"
+)
+
+func init() {
+	if err := measurement.Register(systemPodMetricsName, createSystemPodMetricsMeasurement); err != nil {
+		klog.Fatalf("Cannot register %s: %v", systemPodMetricsName, err)
+	}
+}
+
+func createSystemPodMetricsMeasurement() measurement.Measurement {
+	return &systemPodMetricsMeasurement{}
+}
+
+type systemPodMetricsMeasurement struct{}
+
+type containerMetrics struct {
+	Name         string `json:"name"`
+	RestartCount int32  `json:"restartCount"`
+}
+
+type podMetrics struct {
+	Name       string             `json:"name"`
+	Containers []containerMetrics `json:"containers"`
+}
+
+type systemPodsMetrics struct {
+	Pods []podMetrics `json:"pods"`
+}
```

Comment:
Do we plan to extend this measurement with other metrics in the future? Unless the answer is "yes" I'd use more specific term than metrics (e.g. "restarts").

Reply:
Discussed offline